### ISSUE TITLE
[4.0] Custom fileds category

### ISSUE THREAD

**Issue by Razzo1987** [joomla/joomla-cms#29126]
### Steps to reproduce the issue
Try to create a custom filed and select a category different from "All":
![immagine](https://user-images.githubusercontent.com/906604/82153321-f9028700-9866-11ea-917f-61b6a1e456ec.png)

"All" remain as a text after the category selected

### Expected result

![immagine](https://user-images.githubusercontent.com/906604/82153247-9ad5a400-9866-11ea-9a0a-2719c3ab43b8.png)
OR
![immagine](https://user-images.githubusercontent.com/906604/82153265-aa54ed00-9866-11ea-964a-c37b60d0805f.png)
OR
![immagine](https://user-images.githubusercontent.com/906604/82153282-c8225200-9866-11ea-9cd5-6f453ddaedd9.png)

### Actual result
![immagine](https://user-images.githubusercontent.com/906604/82153332-0ddf1a80-9867-11ea-910e-8642e34830cf.png)

![immagine](https://user-images.githubusercontent.com/906604/82153346-294a2580-9867-11ea-8613-9f37ec247458.png)

### Additional comments

I think the problem is that "All" option have no value

```
<joomla-field-fancy-select  search-placeholder="Type or Select a Category" ><select id="jform_assigned_cat_ids" name="jform[assigned_cat_ids][]" multiple>
	<option value="">All</option>
	<option value="-1" selected="selected">None</option>
	<option value="2">Uncategorised</option>
	<option value="8">Blog</option>
	<option value="9">Help</option>
</select>
```
Also, there shouldn't be able to select a category (or "All") and "none" at the same time:

![immagine](https://user-images.githubusercontent.com/906604/82153400-834aeb00-9867-11ea-896f-a6235af97648.png)

**Comment by chmst** [comment]
confirmed
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29126">issues.joomla.org/tracker/joomla-cms/29126</a>.</sub>

**Comment by Quy** [comment]
Fixed PR #31365
